api/chalicelib/core/feature_flags.py

```diff
@@ -0,0 +1,342 @@
+
+    results = {"total": rows[0]["count"]}
+
+    rows = helper.list_to_camel_case(rows)
+    for row in rows:
+        row.pop("count")
+        row["createdAt"] = TimeUTC.datetime_to_timestamp(row["createdAt"])
+        row["updatedAt"] = TimeUTC.datetime_to_timestamp(row["updatedAt"])
+
+    results["list"] = rows
+    return results
+
+
+def get_feature_flag(project_id: int, feature_flag_id: int) -> Optional[Dict[str, Any]]:
+    """
+    Get a single feature flag by its ID.
+    """
+    feature_flag_columns = (
+        "feature_flag_id",
+        "name",
+        "flag_key",
+        "description",
+        "flag_type",
+        "is_persist",
+        "is_active",
+        "created_at",
+        "updated_at",
+        "created_by",
+        "updated_by",
+    )
+
+    feature_flag_sql = f"""
+        SELECT {", ".join(feature_flag_columns)}
+        FROM feature_flags
+        WHERE feature_flag_id = %(feature_flag_id)s AND project_id = %(project_id)s
+        AND deleted_at IS NULL
+    """
+
+    with pg_client.PostgresClient() as cur:
+        feature_flag_query = cur.mogrify(
+            feature_flag_sql,
+            {"feature_flag_id": feature_flag_id, "project_id": project_id}
+        )
+        cur.execute(feature_flag_query)
+        feature_flag_row = cur.fetchone()
+
+        if feature_flag_row is None:
+            return {"errors": ["Feature flag not found"]}
+
+        feature_flag_row["created_at"] = TimeUTC.datetime_to_timestamp(feature_flag_row["created_at"])
+        feature_flag_row["updated_at"] = TimeUTC.datetime_to_timestamp(feature_flag_row["updated_at"])
+
+    feature_flag_row["conditions"] = get_conditions(feature_flag_id)
+
+    return helper.dict_to_camel_case(feature_flag_row)
+
+
+def create_feature_flag(feature_flag: schemas.FeatureFlagSchema, project_id: int, user_id: int) -> dict:
+    """
+    Create a new feature flag and return its data.
+    """
+    columns = (
+        "feature_flag_id",
+        "name",
+        "flag_key",
+        "description",
+        "flag_type",
+        "is_persist",
+        "is_active",
+        "created_by",
+        "updated_by",
+        "created_at",
+        "updated_at"
+    )
+
+    sql = f"""
+        INSERT INTO feature_flags
+        (project_id, name, flag_key, description, flag_type, is_persist, is_active, created_by)
+        VALUES (%(project_id)s, %(name)s, %(flag_key)s, %(description)s, %(flag_type)s, %(is_persist)s,
+        %(is_active)s, %(created_by)s)
+        RETURNING {", ".join(columns)}
+    """
+
+    with pg_client.PostgresClient() as cur:
+        query = cur.mogrify(sql, {"project_id": project_id, "created_by": user_id, **feature_flag.dict()})
+        cur.execute(query)
+        row = cur.fetchone()
+
+    row["created_at"] = TimeUTC.datetime_to_timestamp(row["created_at"])
+    row["updated_at"] = TimeUTC.datetime_to_timestamp(row["updated_at"])
+    row["conditions"] = create_conditions(row["featureFlagId"], feature_flag.conditions)
```

Comment:
you can create the conditions with the same query that creates the feature_flag; 
because here the feature flag could be created but the conditions could fail